client/js/nonprofits/donate/parseFields/customAmounts/index.ts

```diff
@@ -0,0 +1,22 @@
+// License: LGPL-3.0-or-later
+import JsonStringParser from './JsonStringParser';
+const R = require('ramda');
+const { getDefaultAmounts } = require('../../custom_amounts');
+import { splitParam } from '..';
+
+export type Amount = number | CustomAmount;
+
+export interface CustomAmount {
+  amount: NonNullable<number>;
+  highlight: string;
+}
+
+export default function parseCustomFields(amountsString: string): Amount[] {
```

Comment:
We want the name to be appropriate but we also want to make sure we're returning a detailed CustomAmount after parsing

```suggestion
export default function parseCustomAmounts(amountsString: string): CustomAmount[] { //
```

Reply:
good catch, thanks

Reply:
I'll update the function name, but this should still be returning `Amount[]`, because this is parsing a mixed string like `"20,{amount:30,highlight:'car'},40"` and Amount can be a custom amount or a number.

Reply:
@hejkal my intent is that they we export the expanded object for each of the parsed arguments. Ultimately, `"40"` is the same as `{amount: 40, highlight: false}`

Reply:
Ahh ok, I can do that. It works as is, but I'm happy to make that update

Reply:
Should we do that even if we're given just numbers, like `"20,40,60"` or only if it's a combination?

Reply:
Even if we're given numbers. The idea is to normalize to a more useful object.

Reply:
Ok, sounds good — I'll update it

Reply:
Ok, this is updated